head/firmware/adc/adc_comms.cpp

```diff
@@ -0,0 +1,24 @@
+#include "head/firmware/adc_comms.hpp"
+
+#include "platform_specific_hal_conf.h"
+
+using namespace adc;
+/*
+ * Public Functions
+ */
+ADC::ADC(ADC_interface ADC_intf_instance1, ADC_interface ADC_intf_instance2)
+    : ADC_intf1(ADC_intf_instance1), ADC_intf2(ADC_intf_instance2) {}
+
+auto ADC::get_readings() -> adc::VoltageRead {
+    auto voltage_read = adc::VoltageRead{
+        .z_motor = static_cast<uint16_t>(
+            (static_cast<float>(adc_read_z_motor()) * FULLSCALE_VOLTAGE) /
```

Comment:
This conversion is problematic. `adc_read_z_motor()` returns a `uint16_t` in [0, 4095] and `ADC_FULLSCALE_OUTPUT` is `4095`. This all makes sense. But `FULLSCALE_VOLTAGE` is `3.3f`, and the result is cast back into a `uint16_t`.

 That means that the end result of this conversion can only have the values `3`, `2`, `1`, and `0`, because this conversion is mapping [0, 4095] -> [0.0, 3.3] then making it an int again which will truncate.

One great way to fix this would just be to make `FULLSCALE_VOLTAGE` be the int `3300`. That makes the result of the conversion be an integer number of millivolts (and we should change the name of the struct or the struct members to reflect this).

Reply:
Changed `FULLSCALE_VOLTAGE` to int and calling the. struct `MillivoltsReadings` now

Reply:
Nice!